crates/bevy_math/src/direction.rs

```diff
@@ -143,6 +143,35 @@ impl Dir2 {
     pub const fn as_vec2(&self) -> Vec2 {
         self.0
     }
+
+    /// Performs a spherical linear interpolation between `self` and `rhs`
+    /// based on the value `s`.
+    ///
+    /// This corresponds to interpolating between the two directions at a constant angular velocity.
+    ///
+    /// When `s == 0.0`, the result will be equal to `self`.
+    /// When `s == 1.0`, the result will be equal to `rhs`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_math::Dir2;
+    /// # use approx::RelativeEq;
+    /// #
+    /// let dir1 = Dir2::X;
+    /// let dir2 = Dir2::Y;
+    ///
+    /// let result1 = dir1.slerp(dir2, 1.0 / 3.0);
+    /// assert!(result1.relative_eq(&Dir2::from_xy(0.75_f32.sqrt(), 0.5).unwrap(), f32::EPSILON, f32::EPSILON));
```

Comment:
It might be easier/nicer to write these with `assert_relative_eq!` (or `assert_abs_diff_eq`):
```suggestion
    /// assert_relative_eq!(result1, Dir2::from_xy(0.75_f32.sqrt(), 0.5).unwrap());
```
(and similarly everywhere else)